plugin/src/main/java/com/google/errorprone/xplat/checker/UnnecessaryConcurrentHashMap.java

```diff
@@ -14,56 +14,197 @@
 
 package com.google.errorprone.xplat.checker;
 
-import static com.google.common.collect.ImmutableList.toImmutableList;
+
 import static com.google.errorprone.BugPattern.SeverityLevel.WARNING;
 
 import com.google.errorprone.BugPattern;
 import com.google.errorprone.VisitorState;
 import com.google.errorprone.bugpatterns.BugChecker;
-import com.google.errorprone.bugpatterns.BugChecker.AssignmentTreeMatcher;
 import com.google.errorprone.bugpatterns.BugChecker.NewClassTreeMatcher;
 import com.google.errorprone.bugpatterns.BugChecker.VariableTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
 import com.google.errorprone.matchers.Description;
 import com.google.errorprone.matchers.Matcher;
 import com.google.errorprone.matchers.Matchers;
 import com.google.errorprone.util.ASTHelpers;
-import com.sun.source.tree.AssignmentTree;
-import com.sun.source.tree.ExpressionTree;
 import com.sun.source.tree.NewClassTree;
 import com.sun.source.tree.Tree;
+import com.sun.source.tree.Tree.Kind;
 import com.sun.source.tree.VariableTree;
-import java.beans.Expression;
-import java.util.concurrent.ConcurrentHashMap;
+import com.sun.source.util.TreePathScanner;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.Optional;
 
+/**
+ * Checks for usage of ConcurrentHashMap and suggests the use of Collections.synchronizedMap.
+ * ConcurrentHashMap is not well supported on at least one available platform.
+ */
 @BugPattern(
     name = "UnnecessaryConcurrentHashMap",
-    summary = "Suggests the use of synchronizedMap instead of ConcurrentHashMap",
+    summary = "Suggests the use of Collections.synchronizedMap instead of ConcurrentHashMap.",
     explanation =
-        "ConcurrentHashMap is not well suited for use on Android devices. For this reason,"
-            + " synchronizedMap is suggested to be used in its place for better compatibility.",
+        "ConcurrentHashMap is not well supported on at least one available platform."
```

Comment:
I think it makes more sense to specify that it's for J2ObjC that is the issue particularly around retention.

This allows users who know what they are doing to decided to make an exception for themselves.

Reply:
After talking with Lukhnos I have updated the message to specify that the issue is with iOS, but have not made it more specific than that because the issue has not been verified yet. I can update it later with more details.